src/main/java/org/broadinstitute/consent/http/db/DataAccessRequestDAO.java

```diff
@@ -35,69 +35,94 @@ public interface DataAccessRequestDAO extends Transactional<DataAccessRequestDAO
-            AND LOWER(e.election_type) = 'dataaccess'
-            AND LOWER(v.type) = 'final') final_access_vote ON final_access_vote.reference_id = dar.reference_id
-          WHERE dar.draft = false
-          AND final_access_vote.last_vote = TRUE
-          AND (LOWER(dar.data->>'status') != 'archived' OR dar.data->>'status' IS NULL)
-          AND dar.submission_date > now() - interval '1 year'
-      """)
+  @SqlQuery(
+      """
+              SELECT dar.id, dar.reference_id, dar.collection_id, dar.parent_id,
+                dar.user_id, dar.create_date, dar.sort_date, dar.submission_date, dar.update_date,
+                (regexp_replace(dar.data #>> '{}', '\\\\u0000', '', 'g'))::jsonb AS data,
+                dd.dataset_id
+              FROM data_access_request dar
+              INNER JOIN dar_dataset dd ON dd.reference_id = dar.reference_id AND dd.dataset_id = :datasetId
+              INNER JOIN (
+                SELECT DISTINCT e.reference_id, LAST_VALUE(v.vote)
+                OVER(
+                  PARTITION BY e.reference_id
+                    ORDER BY v.createdate
+                    RANGE BETWEEN
+                      UNBOUNDED PRECEDING AND
+                      UNBOUNDED FOLLOWING
+                ) last_vote
+                FROM election e
+                INNER JOIN vote v ON e.election_id = v.electionid AND v.vote IS NOT NULL
+                WHERE e.dataset_id = :datasetId
+                AND LOWER(e.election_type) = 'dataaccess'
+                AND LOWER(v.type) = 'final') final_access_vote ON final_access_vote.reference_id = dar.reference_id
+              WHERE dar.submission_date > now() - interval '1 year'
+              AND final_access_vote.last_vote = TRUE
+              AND (LOWER(dar.data->>'status') != 'archived' OR dar.data->>'status' IS NULL)
+          """)
   List<DataAccessRequest> findApprovedDARsByDatasetId(@Bind("datasetId") Integer datasetId);
 
+  /**
+   * This query finds submitted DARs based on a date range.  This would be useful if we wanted to
+   * send notifications for "expiring" DARs 30 days before expiration and again at 7 days.
+   *
+   * @param begin Oldest submission date, in seconds since the epoch
+   * @param end   Newest submission date, in seconds since the epoch
+   * @return List of submitted DARs within the date range provided.
+   */
+  @UseRowReducer(DataAccessRequestReducer.class)
+  @SqlQuery(
+      """
+              SELECT dar.id, dar.reference_id, dar.collection_id, dar.parent_id,
+                dar.user_id, dar.create_date, dar.sort_date, dar.submission_date, dar.update_date,
+                (regexp_replace(dar.data #>> '{}', '\\\\u0000', '', 'g'))::jsonb AS data,
+                dd.dataset_id
+              FROM data_access_request dar
+              LEFT JOIN dar_dataset dd ON dd.reference_id = dar.reference_id
+              WHERE dar.submission_date BETWEEN SYMMETRIC to_timestamp(:begin) AND to_timestamp(:end)
```

Comment:
Why is `SYMMETRIC` needed here? Is it possible that `begin` is after `end` instead of before it?

Reply:
Exactly.